nestor_api/api/api_routes/workflow/advance.py

```diff
@@ -0,0 +1,63 @@
+"""Define the workflow advance route."""
+
+from http import HTTPStatus
+
+import nestor_api.lib.config as config_lib
+import nestor_api.lib.workflow as workflow_lib
+from nestor_api.utils.error_handling import non_blocking_clean
+from nestor_api.utils.logger import Logger
+
+
+def advance_workflow(current_step):
+    """Advance the workflow for all requested applications"""
+    Logger.info(
+        {"current_step": current_step},
+        "[/api/workflow/progress/<current_step>] Workflow advance started",
+    )
+
+    try:
+        # Creating a copy of the working configuration directory
+        config_dir = config_lib.create_temporary_config_copy()
+        config_lib.change_environment("staging", config_dir)
```

Comment:
This comes from the current Nestor implementation, it's what makes me think it is kind of confusing. Why would we checkout staging branch specifically, we just want the "default" configuration a.k.a.  something that is not tied to an environment, what do you think ?

Reply:
Ok, yes missed that one. Still the questions persists about this staging branch for the fixed settings